common/Geometry2d/Rect.cpp

```diff
@@ -35,17 +161,33 @@ bool Rect::hit(const Segment& seg) const {
 bool Rect::hit(Point pt) const { return nearPoint(pt, Robot_Radius); }
 
 void Rect::expand(Point p) {
-    pt[0].x() = min(pt[0].x(), p.x());
-    pt[0].y() = min(pt[0].y(), p.y());
-    pt[1].x() = max(pt[1].x(), p.x());
-    pt[1].y() = max(pt[1].y(), p.y());
+    float _minx = minx();
+    float _miny = miny();
+    float _maxx = maxx();
+    float _maxy = maxy();
+
+    pt[0].x() = min(_minx, (float)p.x());
+    pt[0].y() = min(_miny, (float)p.y());
+    pt[1].x() = max(_maxx, (float)p.x());
+    pt[1].y() = max(_maxy, (float)p.y());
```

Comment:
For this and the pad function, you should make it clear in the hpp that this changes which corners pt1 and pt2 are. This makes it bottom left and upper right always.

Reply:
done